base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/HiddenGroupFilter.java

```diff
@@ -0,0 +1,59 @@
+ /*
+ * This file is part of the ISIS IBEX application.
+ * Copyright (C) 2012-2017 Science & Technology Facilities Council.
+ * All rights reserved.
+ *
+ * This program is distributed in the hope that it will be useful.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution.
+ * EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+ * AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+ * OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+ *
+ * You should have received a copy of the Eclipse Public License v1.0
+ * along with this program; if not, you can obtain a copy from
+ * https://www.eclipse.org/org/documents/epl-v10.php or 
+ * http://opensource.org/licenses/eclipse-1.0.php
+ */
+
+package uk.ac.stfc.isis.ibex.ui.blocks.groups;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import uk.ac.stfc.isis.ibex.configserver.displaying.DisplayGroup;
+
+/**
+ * Utility class for filtering out groups which only contain hidden blocks.
+ */
+public final class HiddenGroupFilter {
+    
+    /**
+     * Class only contains static methods, hence private constructor.
+     */
+    private HiddenGroupFilter() {      
+    }
+    
+    /**
+     * Filters out groups that only contain hidden blocks.
+     * 
+     * @param allGroups - a list of all groups
+     * @param showHidden - whether hidden blocks are being shown
+     * @return - a list of groups that should be displayed
+     */
+    public static Collection<DisplayGroup> getVisibleGroups(Collection<DisplayGroup> allGroups, boolean showHidden) {
+        
+        if (showHidden) {
+            return allGroups;
+        }
+        
+        List<DisplayGroup> result = new ArrayList<DisplayGroup>();
+        for (DisplayGroup g : allGroups) {
+            if (g.containsAnyVisibleBlocks()) {
+                result.add(g);
+            }   
+        }
+        return result;
+    }
```

Comment:
In `ComponentFilteredConfiguration`. This kind of operation is done as follows:

```
    public static Collection<Group> filterGroups(Collection<Group> groups) {
        return Lists.newArrayList(Iterables.filter(groups, new Predicate<Group>() {
			@Override
			public boolean apply(Group group) {
				return !group.hasComponent();
			}
		}));
	}
```

Is that better? There is a slight issue of semantics here that if `showHidden` is true, you return the original object (not a copy), if it's `false`, you return a new object. That could potentially cause issues. 

Performance may be slightly better using Iterables.filter because the algorithm can be better optimised on the knowledge of non-mutation, particularly it could be parallelised. Community advice though suggests that it often isn't the case though and this operation isn't computationally intensive anyhow.

The existing code is more maintainable and readable. In Python I'd typically use a list comprehension because I do think they're better, but the boilerplate to use a Java predicate is just too bulky for simple cases like this.